### PR TITLE
Skill Changes

### PR DESCRIPTION
**Pull request by printer16**
Adds three new skills so every job has a relevant skill and adds a janitor preset.
New skills: Corporate Logistics, Mining Operations, and Theology.
Janitor Preset is trained in basic electrical, construction, and devices.
Also refactors the skill system to be much more friendly to work with.
And renames the classifications: Reasonable and Unreasonable so people don't add skills to get into the average range just because it says terrifying.

### CODE REVIEW

**Comment by BurgerLUA** [comment]
Are skills still cosmetic?

**Comment by printer16** [comment]
yes

**Comment by printer16** [comment]
DNM because mistakes are going to be made

**Comment by BurgerLUA** [comment]
The madman!

**Comment by printer16** [comment]
good now

**Comment by BurgerLUA** [comment]
I don't know if this would be good code practice, but I suggest getting the skill system actually working, but don't make it so anything in the game is affected by the skill. I suggest getting an army of people to get that working for you.

**Comment by printer16** [comment]
This PR is completed minus any changes requested. That would be done in a future PR.

File: code/datums/skills/_skill.dm [hunk 1]
```diff
@@ -32,6 +32,21 @@ var/global/list/SKILLS = null
 		if(24 to 1000)
 			return "Unreasonable"
 
+/datum/skillsystem/proc/SkillCheck(var/mob/living/carbon/human/M, var/skill, var/level)
```
Comment (Arrow768) [inline code/datums/skills/_skill.dm]: I see what you did there.

File: code/datums/skills/_skill.dm [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#define	SKILL_NONE  0
+#define	SKILL_BASIC  1
+#define	SKILL_ADEPT  2
+#define	SKILL_EXPERT  3
+
+var/global/list/SKILLS = null
+
+/datum/skillsystem/
+	var/datum/skillsystem/preset/presetfriend = /datum/skillsystem/preset
```
Comment (Lohikar) [inline code/datums/skills/_skill.dm]: Declared type is different from value. Is this intentional?

File: code/datums/skills/_skill.dm [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	
+/datum/skillsystem/skill/
+    var/ID = "none" // ID of the skill, used in code
```
Comment (Lohikar) [inline code/datums/skills/_skill.dm]: Instance variables should be lowercase.

File: code/datums/skills/_skill.dm [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+			if(!(S.name == skill))
+				continue
+			if(M.skills[S.ID] >= level)
+				return 1
+	return 0
+
+/datum/skillsystem/proc/setup_skills()
+	if(SKILLS == null)
+		SKILLS = list()
+		for(var/T in (typesof(/datum/skillsystem/skill/)-/datum/skillsystem/skill))
```
Comment (Lohikar) [inline code/datums/skills/_skill.dm]: `subtypesof()`

File: code/datums/skills/_skill.dm [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+			var/datum/skillsystem/skill/S = new T
+			if(S.ID != "none")
+				if(!SKILLS.Find(S.field))
```
Comment (Lohikar) [inline code/datums/skills/_skill.dm]: SKILLS is assoc here, so you could replace this if and its body with just `LAZYINITLIST(SKILLS[S.field])`.

File: code/datums/skills/_skill.dm [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+/mob/living/carbon/human/proc/GetSkillClass(points)
+	return gskillsystem.CalculateSkillClass(points, age)
+
+/proc/show_skill_window(var/mob/user, var/mob/living/carbon/human/M)
+	if(!istype(M)) return
+	if(SKILLS == null)
+		gskillsystem.setup_skills()
+
+	if(!M.skills || M.skills.len == 0)
```
Comment (Lohikar) [inline code/datums/skills/_skill.dm]: `if (!LAZYLEN(M.skills))`

File: code/datums/skills/_skill.dm [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		user << "There are no skills to display."
```
Comment (Lohikar) [inline code/datums/skills/_skill.dm]: `to_chat()`.

File: code/datums/skills/_skill.dm [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+		return
+
+	var/HTML = "<body>"
```
Comment (Lohikar) [inline code/datums/skills/_skill.dm]: Use list joining.

File: code/global.dm [hunk 9]
```diff
@@ -87,6 +87,8 @@ var/list/adminlog  = list()
 var/Debug2 = 0
 var/datum/debug/debugobj
 
+var/datum/skillsystem/gskillsystem = /datum/skillsystem
```
Comment (Lohikar) [inline code/global.dm]: Declared type is different from value type. Is this intentional?

**Comment by printer16** [comment]
Closing for now.

**Pull request closed without merging**